From 1e7705f81ec6dacc7f14376d526bab968d590767 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@linaro.org>
Date: Thu, 30 Mar 2023 18:49:08 +0200
Subject: [PATCH 0118/1695] plat-stm32mp1: add CFG_STM32MP1_OPTEE_IN_SYSRAM

Adds config switch CFG_STM32MP1_OPTEE_IN_SYSRAM that defines if
STM32MP1 OP-TEE executes in internal RAM with pager.

Switch CFG_STM32MP1_OPTEE_IN_SYSRAM is default disabled. When disabled,
STM32MP15 HUK and user TA support are default disable as external RAM
may not be secure enough to hold platform secrets.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 17586a8868359a67812e8f3d5d4dd5c3f82f3f76 plat-stm32mp1: add CFG_STM32MP1_OPTEE_IN_SYSRAM]
Change-Id: I0f09f1bd9c1d8a1c1fcb52e9ce577fe2a9c6d76a
Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index a095677a7..f60718bf1 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -114,6 +114,7 @@ $(call force,CFG_SECONDARY_INIT_CNTFRQ,n)
 $(call force,CFG_STM32_GPIO,y)
 $(call force,CFG_STM32_VREFBUF,y)
 $(call force,CFG_STM32MP_CLK_CORE,y)
+$(call force,CFG_STM32MP1_OPTEE_IN_SYSRAM,n)
 $(call force,CFG_STM32MP1_SHARED_RESOURCES,n)
 $(call force,CFG_STM32MP13_CLK,y)
 $(call force,CFG_STM32MP13_REGULATOR_IOD,y)
@@ -137,8 +138,17 @@ CFG_HALT_CORES_ON_PANIC ?= y
 CFG_EXTERNAL_DT ?= y
 CFG_STM32_BSEC_SIP ?= y
 CFG_TEE_CORE_NB_CORE ?= 2
-CFG_WITH_PAGER ?= y
 CFG_WITH_SOFTWARE_PRNG ?= y
+CFG_STM32MP1_OPTEE_IN_SYSRAM ?= n
+ifeq ($(CFG_STM32MP1_OPTEE_IN_SYSRAM),y)
+$(call force,CFG_WITH_PAGER,y)
+CFG_STM32MP15_HUK ?= y
+CFG_WITH_USER_TA ?= y
+else
+CFG_STM32MP15_HUK ?= n
+CFG_WITH_PAGER ?= n
+CFG_WITH_USER_TA ?= n
+endif
 endif # CFG_STM32MP15
 
 ifeq ($(CFG_WITH_PAGER),y)
-- 
2.47.2

